content/docs/iac/concepts/options/ignorechanges.md

```diff
@@ -16,6 +16,20 @@ aliases:
 
 The `ignoreChanges` resource option specifies a list of properties that Pulumi will ignore when it updates existing resources. Pulumi ignores a property by using the old value from the state instead of the value provided by the Pulumi program when determining whether an update or replace is needed. Ignored properties will still be used from the program when there is no previous value in the state, most importantly when creating the resource.
 
+## How ignoreChanges works
+
+After the resource is created, Pulumi relies on the last recorded state for every property named in `ignoreChanges`. During a preview or update, Pulumi:
+
+1. Uses the program value when there is no prior state (typically the initial create).
+1. Uses the serialized value from the stack state for any later previews or updates.
+1. Does not automatically read the live value for the property unless you run a refresh.
+
+Because Pulumi reuses the value stored in the state, an external system can safely update the live resource as long as you synchronize the stack before the next update. Run `pulumi refresh` (or `pulumi up --refresh`) to pull the latest provider values into the state. Skipping the refresh step leaves stale values in the state, and Pulumi will continue to send those stale values to the provider during subsequent updates, which can overwrite the drift you meant to preserve.
+
+{{% notes type="warning" %}}
+When you skip `pulumi refresh` (or `pulumi up --refresh`) after `ignoreChanges` has been set, Pulumi keeps using the previous state value when it performs an update. Providers that require full object replacements—such as AWS load balancer listeners where the entire target group array is sent on every update—will receive the stale values from the state and may reset the live configuration.
+{{% /notes %}}
```

Comment:
This is true if and only if you are using `ignoreChanges` to allow non-IaC management of a field. I don't think its fair to assume that all uses of `ignoreChanges` expect external management (and thus need a refresh before update).

Reply:
I can add something here to note that, but it's not that you necessarily _expect_ external management. Your cloud state could have drifted unintentionally and the result will be the same.

Reply:
tldr. Good point. LGTM

---

Consider the following program + state + cloud:

State:
```
{
  "oldInputs": {
    "x": "a"
  }
}
```

Program:
```ts
new example.Resource_X("example", {
  x: "b"
}, { ignoreChanges: [ "x" ] });
```

Cloud (as returned by `Read`):
```
{
  "x": "c"
}
```

You have added `ignoreChanges: [ "x" ]`, so we definitely don't want the result of running anything to set state or the cloud to `"b"`. 

If you want `Resource_X.x` to be cloud managed, then you want to change the state to have `{ "x": "c" }` and you should run a refresh.

If you want `Resource_X.x` to ramain `"a"` (since that is the last program input)... you can't ever run refresh. This isn't really supported <-- _at this point I figured out I was wrong about how refresh interacted with `ignoreChanges`_